### PR TITLE
perf(metrics): record pre-truncation stdout/stderr byte counts on exec_command overflow (#1320)

### LINKED ISSUE #1320: feat(metrics): record pre-truncation stdout/stderr byte counts on exec_command overflow
## Problem

When `exec_command` truncates output (`output_truncated=true`), the JSONL metric record only contains `output_chars` which reflects the post-cap tail preview size, not the original output size. This makes it impossible to make data-driven decisions about whether the 30 KB cap (`MAX_STDOUT_BYTES`) should be raised.

From logs: 3,326 truncated `exec_command` calls in production; `output_chars` p50=5,122, max=25,728 -- these are post-truncation previews, not original sizes.

## Root cause

There is a three-layer truncation cascade:
1. Drain-task level (`exec_runtime.rs` `MAX_DRAIN_STDOUT_BYTES=30_000` / `MAX_DRAIN_STDERR_BYTES=10_000`) -- discards bytes before `handle_output_persist` ever sees the string
2. `handle_output_persist` -- per-stream byte/line cap
3. `format_shell_output_phase` -- combined `SIZE_LIMIT=5_000` byte cap on the formatted response

Capturing raw size at `handle_output_persist` would be misleading: for large outputs the drain cap has already discarded bytes, so the captured size would be indistinguishable from the cap value (~30k).

## Solution

Add unconditional byte counters (`raw_stdout_bytes`, `raw_stderr_bytes`) inside the drain-task closure in `run_with_timeout` that increment on every line received **before** the budget check, not after. Thread them through `ExecutionResult` -> `run_exec_impl` -> `exec_command_impl` -> `MetricEventBuilder` as `stdout_bytes_raw`/`stderr_bytes_raw` (both `Option<u64>`), emitted only when `output_truncated=true` and `timed_out=false`.

Do not add these fields to `ShellOutput` (LLM-visible schema).

## Files

- `crates/aptu-coder/src/tools/exec_runtime.rs` -- drain-task counters, `ExecutionResult`
- `crates/aptu-coder/src/tools/exec_command.rs` -- thread through to metric emission
- `crates/aptu-coder/src/metrics.rs` -- new `Option<u64>` fields + builder methods + update `test_metric_event_new_fields_round_trip`
- `docs/METRICS.md` -- add rows for `stdout_bytes_raw` and `stderr_bytes_raw`

## Out of scope

Layered dual-capture (drain + persist boundary) is a valid fast-follow once this ships and we have real data on which layer is the binding constraint.

### PR DESCRIPTION
**Pull request by clouatre**
## Summary
Record pre-truncation stdout/stderr byte counts on exec_command output overflow. Adds two new optional metrics fields (stdout_bytes_raw, stderr_bytes_raw) to track the actual bytes read before any truncation cap fires, enabling better observability of command output patterns and truncation behavior.

## Changes
- crates/aptu-coder/src/tools/exec_runtime.rs
- crates/aptu-coder/src/tools/exec_command.rs
- crates/aptu-coder/src/metrics.rs
- crates/aptu-coder/src/tests/exec_tests.rs
- crates/aptu-coder/src/tests/metrics_tests.rs
- docs/METRICS.md

## Test plan
- [x] Tests pass (see 03-build.json test_results)
- [x] Linter clean
- [x] Security scan clean (see 04-validation.json security_summary)

Closes #1320

### CODE REVIEW

File: crates/aptu-coder/src/tools/exec_runtime.rs [hunk 1]
```diff
@@ -88,23 +90,28 @@ pub(crate) async fn run_with_timeout(
...
         let mut se_bytes = 0usize;
+        let mut raw_so = 0usize;
+        let mut raw_se = 0usize;
 
         match (so_stream, se_stream) {
             (Some(so), Some(se)) => {
                 let mut merged = so.merge(se);
                 while let Some(Ok((is_stderr, line))) = merged.next().await {
                     let entry_len = line.len() + 1; // +1 for newline
                     if is_stderr {
```
[the remark this answers is not in the document]
Comment (clouatre) [reply crates/aptu-coder/src/tools/exec_runtime.rs:104]: Added a clarifying comment above all three `let entry_len = line.len() + 1;` occurrences in the drain loop explaining that `entry_len` approximates the on-wire byte count (line content + the newline stripped by `LinesStream`), consistent with the `byte_budget_hit` accounting.

File: docs/METRICS.md [hunk 2]
```diff
@@ -45,8 +45,9 @@ Each line in the JSONL file is one JSON object:
...
-| `filter_applied` | `string \| null` | Name of the filter rule that matched and transformed the output (e.g., `"git pull"`, `"cargo build"`); `null` when no filter fired or for non-`exec_command` tools. Omitted from JSONL when `null` (`#[serde(skip_serializing_if)]`). |
 | `chars_threshold_breach` | `bool` | `true` when `output_chars > 30,000`; fires for the top ~0.33% of `exec_command` calls (p99.7 of 27,981 observed calls). Early-warning signal for responses approaching the per-stream byte-cap threshold (MAX_STDOUT_BYTES = 30,000). Omitted from JSONL when `false` (`#[serde(skip_serializing_if)]`); defaults to `false` on parse for backward compatibility. |
+| `stdout_bytes_raw` | `u64 \| null` | Raw stdout bytes read before any truncation; populated only when `output_truncated=true` and `timed_out=false`. Omitted from JSONL when `null` (`#[serde(skip_serializing_if)]`). |
+| `stderr_bytes_raw` | `u64 \| null` | Raw stderr bytes read before any truncation; populated only when `output_truncated=true` and `timed_out=false`. Omitted from JSONL when `null` (`#[serde(skip_serializing_if)]`). |
```
[the remark this answers is not in the document]
Comment (clouatre) [reply docs/METRICS.md]: Fixed. Added `filter_applied` as a row in the primary metrics field table (immediately after `exit_code`), making it consistent with the schema and removing the inconsistency between the primary table and the backward-compat section.

File: crates/aptu-coder/src/tools/exec_command.rs [hunk 3]
```diff
@@ -555,25 +555,29 @@ pub(crate) async fn exec_command_impl(
...
+        .stdin_provided(stdin_provided)
+        .timeout_configured_ms(timeout_configured_ms)
+        .drain_timeout_ms(drain_timeout_ms)
+        .working_dir_used(working_dir_used);
+    // Emit raw byte counts only when output was truncated and not timed out.
+    if output_truncated && !output.timed_out {
+        metric_builder = metric_builder
+            .stdout_bytes_raw(raw_stdout_bytes)
+            .stderr_bytes_raw(raw_stderr_bytes);
+    }
```
[the remark this answers is not in the document]
Comment (clouatre) [reply crates/aptu-coder/src/tools/exec_command.rs]: Fixed. Added `&& output.output_collection_error.is_none()` to the guard, so `stdout_bytes_raw`/`stderr_bytes_raw` are only emitted when output was byte-budget truncated -- not when the drain task aborted (counters forced to 0 in that path). Updated the METRICS.md descriptions for both fields to reflect the corrected condition.

File: crates/aptu-coder/src/tests/exec_tests.rs [hunk 4]
```diff
@@ -384,3 +384,76 @@ async fn test_persist_interleaved_mid_char_boundary() {
     );
     let _char_count = preview.chars().count();
 }
+
+#[tokio::test]
+async fn test_run_exec_impl_raw_byte_counters() {
+    // Edge case: unconditional byte counters increment on every line received
+    // even after budget check fires. Use a command that produces output exceeding
+    // the drain budget to verify raw counters exceed the budget.
```
[the remark this answers is not in the document]
Comment (clouatre) [reply crates/aptu-coder/src/tests/exec_tests.rs]: Fixed. Corrected the comment to accurately describe what the test does: verifies raw byte counters are populated for normal (non-truncated) output. The separate test `test_run_exec_impl_raw_counters_exceed_budget` covers the budget-overflow case.
